security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -485,12 +488,16 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 		return nil, err
 	}
 
+	backOffInMilliSec := rand.Int63n(sc.configOptions.InitialBackoff)
+	log.Debugf("Wait for %d millisec for initial CSR", backOffInMilliSec)
+	// Add a jitter to initial CSR to avoid thundering herd problem.
+	time.Sleep(time.Duration(backOffInMilliSec) * time.Millisecond)
```

Comment:
What are the implications of blocking in this thread?

Reply:
This is to add some jitter before making the CSR request to avoid the thundering herd problem.

Reply:
i meant whether or not it was safe to call sleep and block the thread, but after looking the whole thing over in more detail, looks like everything is entirely synchronous so adding a few sleeps should be fine.